.github/workflows/infra_apply.yaml

```diff
@@ -0,0 +1,174 @@
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: Environment where the resources will be deployed.
+        type: string
+        required: true
+      region:
+        description: Azure region where the resources will be deployed.
+        type: string
+        required: true
+      base_path:
+        description: The base path on which the script will look for Terraform projects
+        type: string
+        required: true
+      env_vars:
+        description: List of environment variables to set up, given in env=value format.
+        type: string
+        required: false
+      use_private_agent:
+        description: Use a private agent to run the Terraform plan.
+        type: boolean
+        required: false
+        default: false
+
+env:
+  ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+  ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+  ARM_USE_OIDC: true
+  ARM_USE_AZUREAD: true
+  ARM_STORAGE_USE_AZUREAD: true
+
+concurrency:
+  group: ${{ github.workflow }}-cd
+  cancel-in-progress: true
+
+jobs:
+
+  tf_plan:
+    name: 'Terraform Plan'
+    runs-on: ${{ inputs.use_private_agent == true && 'self-hosted' || 'ubuntu-20.04' }}
+    environment: ${{ inputs.environment }}-ci
+    permissions:
+      id-token: write
+      contents: read
+    env:
+      ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+    outputs:
+      working_dir: ${{ steps.directory.outputs.dir }}
+      terraform_version: ${{ steps.set-terraform-version.outputs.terraform_version }}
+
+    steps:
+
+      # Set the directory where the Terraform files are located
+      # The directory the value is then available in ${{ steps.directory.outputs.dir }}
+      - name: Set directory
+        id: directory
+        run: |
+          set -e
+          if [ -z "${{ inputs.environment }}" ] || [ -z "${{ inputs.region }}" ]; then
+            echo "Both environment and region must be provided."
+            exit 1
+          else
+            # The directory is expected to be in the format
+            #  infra/resources/${{ inputs.environment }}/${{ inputs.region }}
```

Comment:
```suggestion
            #  ${inputs.base_path}/${{ inputs.environment }}/${{ inputs.region }}
```

(?)

Reply:
imho this comment is not necessary, the code is already clear enough. I would just leave the example (infra/westeurope ...)